Public/Sdk/SelfHost/BuildXL/Testing/QTest/qtestFramework.dsc

```diff
@@ -62,16 +62,27 @@ function runMultipleQTests(args: TestRunArguments) : File[]
     ];
 }
 
-function runTest(args : TestRunArguments) : File[] {
-
-    if (args.limitGroups && args.limitGroups.length > 1) {
-        Contract.fail("Multiple limitGroups should be broken down in runMultipleQTests() before calling this function.");
-    }
-
+function runTest(args : TestRunArguments) : File[] {  
+    const testMethod = Environment.getStringValue("[UnitTest]Filter.testMethod");
+    const testClass  = Environment.getStringValue("[UnitTest]Filter.testClass");
+     
+    let additionalOptions = undefined;
+    let filterArgs = [];
     let rootTestAdapterPath = importFrom("xunit.runner.visualstudio").Contents.all;
     let testAdapterPath =  d`${rootTestAdapterPath}/build/_common`;
-
-    if (args.parallelGroups && args.parallelGroups.length > 0) {
+    
+    // when testmethod or testclass ignore limitGroups and skipGroups arguments
+    if (testMethod || testClass) {
+        // vstest doesn't support a class filter for xunit runner. 
+        // reuse the FullyQualifiedName to filter the testClass, which is not an exactly match. 
+        // if a class's FullyQualifiedName contains another class's FullyQualifiedName, both classes' test cases will be run.
+        // user can add "." at the end of the testclass to get the exact match.
+        let methodAndClassfilterArgs = [
+            ...(testMethod ? ["FullyQualifiedName="+testMethod,] : []),
+            ...(testClass ? ["FullyQualifiedName="+testClass,] : []),
+        ];
+        filterArgs = [methodAndClassfilterArgs.join("|")];
+    } else if (args.parallelGroups && args.parallelGroups.length > 0) {

```

Comment:
Can't you have both "testClass or testMethod" **and** parallel groups, in which case I would expect both to be added to `filterArgs`?